### PR TITLE
Add script for training a branch predictor

### PR DESCRIPTION
**Pull request by nkanderson**
This script will process a training data file in chunks specified by the `batch_size` argument. The following shows output from running this on a 517MB csv training file, with an input size of 30, hidden layer size of 2, 500 epochs, learning rate of 0.2, and batch size of 4096.

@tnl3pdx let me know if you want to take a look at this one before I merge.

```sh
boolean-predictor-MLP git:(main) ✗ time ./bin/train_bp mlp_training_data_20251125_102015.csv 30 2 500 0.2 4096
Creating MLP with:
  Input size: 30
  Hidden layer size: 2
  Batch size: 4096

Training from: mlp_training_data_20251125_102015.csv
Epochs: 500, Learning rate: 0.2

Starting training...
Epoch 1/500 - 23397542 samples processed
Epoch 50/500
Epoch 100/500
Epoch 150/500
Epoch 200/500
Epoch 250/500
Epoch 300/500
Epoch 350/500
Epoch 400/500
Epoch 450/500
Epoch 500/500

Training complete!
Total samples per epoch: 23397542

Saving weights...
Weights saved to: mlp_30_2.txt
./bin/train_bp mlp_training_data_20251125_102015.csv 30 2 500 0.2 4096  7778.50s user 92.26s system 94% cpu 2:18:39.32 total
```

### CODE REVIEW

**Review by tnl3pdx — APPROVED** [review]
Looks great! I ran the script with a ~20 MB file with the same parameters and got a weight table output from it. 
image
